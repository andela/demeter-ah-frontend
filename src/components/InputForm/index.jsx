import React from 'react';
import Input from '../Input';
import Label from '../Label';

const InputForm = ({
  labelname, name, inputType,
  classes, inputClass, labelClass, ...rest
}) => (
  <div className={classes}>
    <Label labelname={labelname} name={name} classes={labelClass} />
    <Input inputType={inputType} name={name} classes={inputClass} {...rest} />
  </div>
);

export default InputForm;
