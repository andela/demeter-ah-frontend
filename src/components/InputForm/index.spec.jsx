import React from 'react';
import { shallow, mount } from 'enzyme';
import InputForm from './index';

describe('InputForm Component', () => {
  it('Should render without errors in debug mode', () => {
    const component = shallow(<InputForm debug />);
    expect(component).toMatchSnapshot();
  });

  it('Should render along with children componenet', () => {
    const component = mount(<InputForm />);
    expect(component).toMatchSnapshot();
    expect(component.find('input')).toHaveLength(1);
    expect(component.find('label')).toHaveLength(1);
  });
});
