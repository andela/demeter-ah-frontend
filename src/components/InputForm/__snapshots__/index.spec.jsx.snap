// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm Component Should render along with children componenet 1`] = `
<InputForm>
  <div>
    <Label>
      <label />
    </Label>
    <Input>
      <input
        className="forminput rounded py-2 bg-purple-150 undefined"
        required={true}
      />
    </Input>
  </div>
</InputForm>
`;

exports[`InputForm Component Should render without errors in debug mode 1`] = `
<div>
  <Label />
  <Input
    debug={true}
  />
</div>
`;
